File: core/src/no/elg/infiniteBootleg/world/generator/EmptyChunkGenerator.java
package no.elg.infiniteBootleg.world.generator;

import no.elg.infiniteBootleg.world.Chunk;
import no.elg.infiniteBootleg.world.World;
import no.elg.infiniteBootleg.world.generator.biome.Biome;
import org.jetbrains.annotations.NotNull;

/**
 * Generate chunks that are always air
 *
 * @author Elg
 */
public class EmptyChunkGenerator implements ChunkGenerator {

    @Override
    public @NotNull Biome getBiome(int worldX) {
        return Biome.PLAINS;
    }

    @Override
    public int getHeight(int worldX) {
        return 0;
    }

    @Override
    public @NotNull Chunk generate(@NotNull World world, int chunkX, int chunkY) {
        return new Chunk(world, chunkX, chunkY);
    }
}

File: core/src/main/java/no/elg/infiniteBootleg/world/generator/EmptyChunkGenerator.kt
package no.elg.infiniteBootleg.world.generator

import no.elg.infiniteBootleg.world.Chunk
import no.elg.infiniteBootleg.world.ChunkImpl
import no.elg.infiniteBootleg.world.generator.biome.Biome
import no.elg.infiniteBootleg.world.world.World

/**
 * Generate chunks that are always air
 *
 * @author Elg
 */
class EmptyChunkGenerator : ChunkGenerator {
  override fun getBiome(worldX: Int): Biome {
    return Biome.PLAINS
  }

  override fun getHeight(worldX: Int): Int {
    return 0
  }

  override fun generate(world: World, chunkX: Int, chunkY: Int): Chunk {
    val chunk = ChunkImpl(world, chunkX, chunkY)
    chunk.finishLoading()
    return chunk
  }
}
